core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -732,6 +736,9 @@ private[deploy] class Worker(
           trimFinishedExecutorsIfNecessary()
           coresUsed -= executor.cores
           memoryUsed -= executor.memory
+          if (CLEANUP_NON_SHUFFLE_FILES_ENABLED) {
+            shuffleService.executorRemoved(executorStateChanged.execId.toString, appId)
```

Comment:
Re-reading this code with fresh eyes made me realize that maybe the method naming here is a bit confusing: from the name alone, it sounds like `executorRemoved` is a notification-sending method, the kind of thing where you'd always want to call it when the corresponding event (executor removal) occurs and then leave it up to the method implementation to decide whether to actually carry out specific cleanup logic.

On the other hand, we call `shuffleService.applicationRemoved()` in `maybeCleanupApplication()` and there's a flag in that method controlling whether we actually clean up, so we have a similar pattern there.

Reply:
It takes me great pain to name a function, config, etc, so I just followed the existing similar pattern. Thus, I'm happy to rename them, perhaps `cleanupNonshuffleFilesOfExecutor`?

Reply:
Nah, no need to rename since this matches existing convention. I was just thinking aloud here / explaining the reasoning for other future readers.